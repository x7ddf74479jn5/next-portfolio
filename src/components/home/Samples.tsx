import Headline from "src/components/home/Headline";
import HorizontalRule from "src/components/home/HorizontalRule";
import SampleCarousel from "src/components/home/SampleCarousel";
import { WrapInSection } from "src/components/layouts/Section";
import Spacer from "src/components/layouts/Spacer";

import { samples } from "../../contents/samples";
import { NextArrowButton } from "../common/NextArrowButton";

const Samples = () => {
  return (
    <WrapInSection>
      <Headline>Samples</Headline>
      <HorizontalRule />
      <Spacer size="sm" />
      <Spacer size="md" />
      <SampleCarousel samples={samples} />
      <Spacer size="md" />
      <NextArrowButton label="全ての事例を見る" href="/samples/" />
    </WrapInSection>
  );
};

export default Samples;
