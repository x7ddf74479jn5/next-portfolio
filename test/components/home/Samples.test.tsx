import Samples from "src/components/home/Samples";
import type { RenderResult } from "test/test-utils";
import { render } from "test/test-utils";

describe("Samples", () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(<Samples />);
  });

  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    expect(renderResult.getByRole("heading", { level: 2 })).toHaveTextContent("Samples");
    expect(renderResult.getByRole("separator")).toBeInTheDocument();
    expect(renderResult.getByRole("button")).toHaveTextContent("全ての事例を見る");
    expect(renderResult.getByRole("button")).toHaveAttribute("href", expect.stringMatching("samples"));
  });
});
