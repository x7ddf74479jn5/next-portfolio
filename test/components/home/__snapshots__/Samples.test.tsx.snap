// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Samples matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="section"
  >
    <div
      class="sectionWrapIn"
    >
      <h2
        class="headlineTop"
      >
        Samples
      </h2>
      <hr
        class="sectionLine"
      />
      <div
        class="sm"
      />
      <div
        class="md"
      />
      <div
        class="swiper-container swiper-container-initialized swiper-container-horizontal"
      >
        <div
          class="swiper-button-prev"
        />
        <div
          class="swiper-button-next"
        />
        <div
          class="swiper-pagination swiper-pagination-clickable swiper-pagination-bullets"
        />
        <div
          class="swiper-wrapper"
          style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
        >
          <div
            class="swiper-slide swiper-slide-duplicate"
            data-swiper-slide-index="4"
          >
            <article
              class="gridListItem"
            >
              <a
                class="textLineNone"
                href="/samples/ec"
                target="_self"
              >
                <h3>
                  ECサイト(開発中)
                </h3>
                <div
                  class="mediaThumb"
                >
                  <div
                    style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
                  >
                    <img
                      alt="eye catch of EC app"
                      data-nimg="true"
                      decoding="async"
                      sizes="100vw"
                      src="/_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=3840&q=75"
                      srcset="/_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=640&q=75 640w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=750&q=75 750w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=828&q=75 828w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=3840&q=75 3840w"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                    />
                  </div>
                </div>
                <p>
                  React学習用に開発しているデモサイトです。
                </p>
              </a>
            </article>
          </div>
          <div
            class="swiper-slide"
            data-swiper-slide-index="0"
          >
            <article
              class="gridListItem"
            >
              <a
                class="textLineNone"
                href="/samples/food-blog"
                target="_self"
              >
                <h3>
                  Food Blog
                </h3>
                <div
                  class="mediaThumb"
                >
                  <div
                    style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
                  >
                    <img
                      alt="eye catch of food-blog"
                      data-nimg="true"
                      decoding="async"
                      sizes="100vw"
                      src="/_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=3840&q=75"
                      srcset="/_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=640&q=75 640w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=750&q=75 750w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=828&q=75 828w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=3840&q=75 3840w"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                    />
                  </div>
                </div>
                <p>
                  Next.js, MDX, microCMSを使ったレシピブログです。
                </p>
              </a>
            </article>
          </div>
          <div
            class="swiper-slide"
            data-swiper-slide-index="1"
          >
            <article
              class="gridListItem"
            >
              <a
                class="textLineNone"
                href="/samples/portfolio"
                target="_self"
              >
                <h3>
                  ポートフォリオ
                </h3>
                <div
                  class="mediaThumb"
                >
                  <div
                    style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
                  >
                    <img
                      alt="eye catch of portfolio"
                      data-nimg="true"
                      decoding="async"
                      sizes="100vw"
                      src="/_next/image?url=%2Fimg%2Fsamples%2Fportfolio-eyecatch.webp&w=3840&q=75"
                      srcset="/_next/image?url=%2Fimg%2Fsamples%2Fportfolio-eyecatch.webp&w=640&q=75 640w, /_next/image?url=%2Fimg%2Fsamples%2Fportfolio-eyecatch.webp&w=750&q=75 750w, /_next/image?url=%2Fimg%2Fsamples%2Fportfolio-eyecatch.webp&w=828&q=75 828w, /_next/image?url=%2Fimg%2Fsamples%2Fportfolio-eyecatch.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimg%2Fsamples%2Fportfolio-eyecatch.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimg%2Fsamples%2Fportfolio-eyecatch.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimg%2Fsamples%2Fportfolio-eyecatch.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimg%2Fsamples%2Fportfolio-eyecatch.webp&w=3840&q=75 3840w"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                    />
                  </div>
                </div>
                <p>
                  現在閲覧してくださっているサイトです。
                </p>
              </a>
            </article>
          </div>
          <div
            class="swiper-slide"
            data-swiper-slide-index="2"
          >
            <article
              class="gridListItem"
            >
              <a
                class="textLineNone"
                href="/samples/my-favorite-gear"
                target="_self"
              >
                <h3>
                  My Favorite Gear
                </h3>
                <div
                  class="mediaThumb"
                >
                  <div
                    style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
                  >
                    <img
                      alt="eye catch of my-favorite-gear"
                      data-nimg="true"
                      decoding="async"
                      sizes="100vw"
                      src="/_next/image?url=%2Fimg%2Fsamples%2Fmy-favorite-gear-eyecatch.webp&w=3840&q=75"
                      srcset="/_next/image?url=%2Fimg%2Fsamples%2Fmy-favorite-gear-eyecatch.webp&w=640&q=75 640w, /_next/image?url=%2Fimg%2Fsamples%2Fmy-favorite-gear-eyecatch.webp&w=750&q=75 750w, /_next/image?url=%2Fimg%2Fsamples%2Fmy-favorite-gear-eyecatch.webp&w=828&q=75 828w, /_next/image?url=%2Fimg%2Fsamples%2Fmy-favorite-gear-eyecatch.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimg%2Fsamples%2Fmy-favorite-gear-eyecatch.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimg%2Fsamples%2Fmy-favorite-gear-eyecatch.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimg%2Fsamples%2Fmy-favorite-gear-eyecatch.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimg%2Fsamples%2Fmy-favorite-gear-eyecatch.webp&w=3840&q=75 3840w"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                    />
                  </div>
                </div>
                <p>
                  自分のお気に入りのアイテムを8つ選んでTwitterに投稿できます。
                </p>
              </a>
            </article>
          </div>
          <div
            class="swiper-slide"
            data-swiper-slide-index="3"
          >
            <article
              class="gridListItem"
            >
              <a
                class="textLineNone"
                href="/samples/chat"
                target="_self"
              >
                <h3>
                  チャットボット
                </h3>
                <div
                  class="mediaThumb"
                >
                  <div
                    style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
                  >
                    <img
                      alt="eye catch of chat"
                      data-nimg="true"
                      decoding="async"
                      sizes="100vw"
                      src="/_next/image?url=%2Fimg%2Fsamples%2Fchat-bot-eyecatch.webp&w=3840&q=75"
                      srcset="/_next/image?url=%2Fimg%2Fsamples%2Fchat-bot-eyecatch.webp&w=640&q=75 640w, /_next/image?url=%2Fimg%2Fsamples%2Fchat-bot-eyecatch.webp&w=750&q=75 750w, /_next/image?url=%2Fimg%2Fsamples%2Fchat-bot-eyecatch.webp&w=828&q=75 828w, /_next/image?url=%2Fimg%2Fsamples%2Fchat-bot-eyecatch.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimg%2Fsamples%2Fchat-bot-eyecatch.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimg%2Fsamples%2Fchat-bot-eyecatch.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimg%2Fsamples%2Fchat-bot-eyecatch.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimg%2Fsamples%2Fchat-bot-eyecatch.webp&w=3840&q=75 3840w"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                    />
                  </div>
                </div>
                <p>
                  事前に用意したQAを使って自動応答するチャットボットです。
                </p>
              </a>
            </article>
          </div>
          <div
            class="swiper-slide"
            data-swiper-slide-index="4"
          >
            <article
              class="gridListItem"
            >
              <a
                class="textLineNone"
                href="/samples/ec"
                target="_self"
              >
                <h3>
                  ECサイト(開発中)
                </h3>
                <div
                  class="mediaThumb"
                >
                  <div
                    style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
                  >
                    <img
                      alt="eye catch of EC app"
                      data-nimg="true"
                      decoding="async"
                      sizes="100vw"
                      src="/_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=3840&q=75"
                      srcset="/_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=640&q=75 640w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=750&q=75 750w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=828&q=75 828w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimg%2Fsamples%2Fec-app-eyecatch.webp&w=3840&q=75 3840w"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                    />
                  </div>
                </div>
                <p>
                  React学習用に開発しているデモサイトです。
                </p>
              </a>
            </article>
          </div>
          <div
            class="swiper-slide swiper-slide-duplicate"
            data-swiper-slide-index="0"
          >
            <article
              class="gridListItem"
            >
              <a
                class="textLineNone"
                href="/samples/food-blog"
                target="_self"
              >
                <h3>
                  Food Blog
                </h3>
                <div
                  class="mediaThumb"
                >
                  <div
                    style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
                  >
                    <img
                      alt="eye catch of food-blog"
                      data-nimg="true"
                      decoding="async"
                      sizes="100vw"
                      src="/_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=3840&q=75"
                      srcset="/_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=640&q=75 640w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=750&q=75 750w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=828&q=75 828w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimg%2Fsamples%2Ffood-blog-eyecatch.webp&w=3840&q=75 3840w"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
                    />
                  </div>
                </div>
                <p>
                  Next.js, MDX, microCMSを使ったレシピブログです。
                </p>
              </a>
            </article>
          </div>
        </div>
      </div>
      <div
        class="md"
      />
      <a
        class="btnRound iconArrowNext"
        href="/samples"
        role="button"
      >
        全ての事例を見る
      </a>
    </div>
  </section>
</DocumentFragment>
`;
